src/systems/filecoin_vm/actor/actor.go

```diff
@@ -95,3 +97,72 @@ func ArgPop(params *MethodParams, rt Has_AbortArg) Serialization {
 func ArgEnd(params *MethodParams, rt Has_AbortArg) {
 	CheckArgs(params, rt, len(*params) == 0)
 }
+
+// Create a new entry in the balance table, with the specified initial balance.
+// May fail if the specified address already exists in the table.
+func BalanceTable_WithNewAddressEntry(table BalanceTableHAMT, address addr.Address, initBalance TokenAmount) (
+	ret BalanceTableHAMT, ok bool) {
+
+	IMPL_FINISH()
+	panic("")
+}
+
+// Delete the specified entry in the balance table.
+// May fail if the specified address does not exist in the table.
+func BalanceTable_WithDeletedAddressEntry(table BalanceTableHAMT, address addr.Address) (
+	ret BalanceTableHAMT, ok bool) {
+
+	IMPL_FINISH()
+	panic("")
+}
+
+// Add the given amount to the given address's balance table entry.
+func BalanceTable_WithAdd(table BalanceTableHAMT, address addr.Address, amount TokenAmount) (
+	ret BalanceTableHAMT, ok bool) {
+
+	IMPL_FINISH()
+	panic("")
+}
+
+// Subtract the given amount (or as much as possible, without making the resulting amount negative)
+// from the given address's balance table entry, independent of any minimum balance maintenance
+// requirement.
+// Note: ok should be set to true here, even if the operation caused the entry to hit zero.
+// The only failure case is when the address does not exist in the table.
+func BalanceTable_WithSubtractPreservingNonnegative(
```

Comment:
Yes, it's a special case (and one can be implemented in terms of the other). Separated out mainly as documentation, since the call sites will look very different (in the case of `Extract`, an untrusted principal is typically extracting as much as possible subject to some limit; whereas in the case of `Subtract`, it's typically a trusted system principal subtracting unconditionally).

Reply:
I see, this explanation is helpful. thanks Joe!

Reply:
When would you ever want to substract more than is available? Seems like an error... no?

Reply:
You wouldn't subtract more than exists in the table, but you might in some cases want to set a separate limit (e.g., a locked balance), so that `Extract` actually subtracts less than exists in the table.